### ISSUE TITLE
[proposal] Use Braille markup to show formatting

### ISSUE THREAD

**Issue by vlegoff** [nvaccess/nvda#8436]
> Note: I wasn't sure filling out an issue for a proposal was the way to do, though it's acceptable in most open-source projects I've contributed to.

I propose that formatting information are optionally provided in Braille.  Currently, there is no consistent way for a user relying only on Braille to guess most formatting information.  I propose something similar to Braille indicators used in web pages to indicate links or landmarks.

### How to indicate formatting changes?

Obviously, not all formatting information can be easily displayed in Braille.  I would propose something like the following:

- `<X>` to indicate the beginning of a formatting change.
- `</X>` to indicate the end of such a formatting change.

Capital and lowercase letters can be alternated too (like `<u>` to indicate the beginning of a underlined section, then `<U>` to indicate the end of the formatted section.

### Which information to display?

The information to display in Braille should be configurable, much as the formatting announcements can be configured.  I would suggest an ability to indicate:

- Underline, bold, and italic.
- Spelling errors.

This system would be easy to extend, adding new information.  Indicating the font, for instance, might not be ideal without some adaptation, but perhaps a change in font could be indicated.

### Example to be displayed in Braille

With "display formatting in Braille" turned on with "bold", in a word document with the following line:

This is a **noun**.

The Braille display could render this line as:

    This is a <b>noun</b>.

Or, depending on the formatting strategy:

    This is a <b>noun<B>.

(I do not propose you implement both strategies, just choose the most consistent one.)

### Various considerations

- The way to display tags should be open to translation, much like HTML indicators, as `<b>` in my example clearly refers to "bold" and would not be as obvious for all languages.
- This will obviously generate an additional shift between Braille cells and content, which is already handled by Web indicators I believe.  The routing cursor will, by definition, be affected as well.

**Comment by josephsl** [comment]
Hi,

This depends on a number of factors:

1. Braille table in use: not all tables support all possible punctuations, and some tables (including Unified English Braille) do come with emphasis indicators.
2. Not all text area objects support various emphasis: different programs may have different ideas about which emphasis to show, or may show the same emphasis indicator for many different things.

There is an add-on called Braille Extender that includes what you might be looking for. I advise talking to developers of that add-on first so your concerns can be addressed from that add-on first. Once the community says that work done by add-on authors is useful for the entire community, then we'll consider this issue (or a possible earlier issue, in which case this issue would stay open if this one provides more info).

Thanks.

**Comment by vlegoff** [comment]
I've suggested this idea to BrailleExternder's developers.  Feel free to close this issue here, or let it open, as you wish!

Thanks again,

**Comment by zstanecic** [comment]
hi,
we cannot use language dependent brailling methods, as some languages 
hasn't revised their braille standards for a long years.

**Comment by derekriemer** [comment]
[**@zstanecic**](https://github.com/zstanecic) commented on [Jun 26, 2018, 3:49 PM MDT](https://github.com/nvaccess/nvda/issues/8436#issuecomment-400474303 "2018-06-26T21:49:56Z - Replied by Github Reply Comments"):
> hi,
> we cannot use language dependent brailling methods, as some languages 
> hasn't revised their braille standards for a long years.

Than we need a method for doing this in those languages, however, in languages with symbols for things like italic we should use them to be consistent.

**Comment by zstanecic** [comment]
Yes, but some languages doesn't have the specifications for italic or 
underline or bold.

What do you plan with then?

**Comment by vlegoff** [comment]
Actually, the Braille table handling this would be (in my opinion) a false solution.  I was thinking more along the lines of the Braille landmarks NVDA provides for links, headings and other things in the Browser.  That's not table specific, though translators can change the landmarks for specific languages.  The problem with table dependent symbols is that, for instance, comp8 braille wouldn't be affected (defining these Braille landmarks in comp8 has no meaning and most specifications don't including a special handling, that will remain true in most cases, I believe).  Whether a language (but not table) -specific landmark comprised of one/two cells (probably 2-4 to delimit the portion in bold/italic/underline) would be much more portable, in my opinion.
